uxarray/core/dataarray.py

```diff
@@ -0,0 +1,35 @@
+import numpy as np
+import xarray as xr
+
+from typing import Optional
+
+from uxarray.core.grid import Grid
+
+
+class UxDataArray(xr.DataArray):
+
+    _uxgrid = None
+
+    def __init__(self, *args, uxgrid: Grid = None, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.uxgrid = uxgrid
+        # TODO: Weird that below if-check leads miscreation of UxDataArray object
```

Comment:
Can you provide a MVCE that lets it fail?
For me this seems to work fine.

Reply:
> Can you provide a MVCE that lets it fail? For me this seems to work fine.

@erogluorhan After uncommenting the section, I tried running this function to see the error:

https://github.com/UXARRAY/uxarray/blob/5082ca28d961923cf57935dc8192f1e5a5b910f1/test/test_dataset.py#L40

The problem is https://github.com/UXARRAY/uxarray/blob/prototype_redesign/uxarray/core/dataset.py#L135 

`self[var_key].to_numpy()
`This following above calls the dataarray constructor with the uxgrid object set to None and so the assertion is triggered.

Reply:
This should be fixed if you use your own implementation of `_replace`

Reply:
Thanks for the pointer, took a little while to fix the root of this. `_replace` wasn't needed, my last commit fixes it.  It was a bug in `__setitem__` function.

Reply:
You are still going to need it. Many instance methods are using it, like `isel`, `rename`, `transpose` etc.

Reply:
Right. Thanks for the details

Reply:
`_replace()` and many other required overridings have been made.